### PR TITLE
KOGITO-4021: BPMN Editor - Fixed client-side marshaller issues during namespaces and schemaLocation processing

### PR DESCRIPTION
**Pull request by romartin**
Hey @hasys @domhanak 

Here is the fix for https://issues.redhat.com/browse/KOGITO-4021, which is a blocker, so plz consider giving this PR priority.

Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
jenkins do fdb

**Comment by hasys** [comment]
Hi @romartin, fix looks good to me, I can compile project now. Great catch, thank you!

But there are two issues I can see:
1. there are failing Selenium tests and it looks like they failing due to changes in this PR. [see Log 1 below]
2. build is successful but there are still some errors in the console log during compilation. Since it is no longer prevent user from using the process I think it is not a blocker and can be managed by separate PR, but I suggest to give it some priority as well. I did a quick look and I think we fixed similar issue in the Business Central some time ago, probably we didn't port the fix to the Kogito. Ping me if you think I should continue investigations.

<details>
  <summary>Log 1 failing Selenium tests</summary>
  <pre>
Test Result (2 failures / +2)

    org.kie.workbench.common.stunner.kogito.client.selenium.BPMNDesignerKogitoSeleniumIT.testBasicModel
    org.kie.workbench.common.stunner.kogito.client.selenium.BPMNDesignerKogitoSeleniumIT.testNewDiagram

Expecting:
 <control instance> and <test instance> to be identical
Expected number of attributes '3' but was '4' - comparing <bpmn2:definitions...> at /definitions[1] to <bpmn2:definitions...> at /definitions[1] expected:<...w.jboss.org/drools" [exporter="jBPM Process Modeler" exporterVersion="2.0" id="_hImTMHArEDiY0b-88EQFvg" targetNamespace="http://www.omg.org/bpmn20">
  <bpmn2:interface id="_E486B83F-1225-436D-94D5-C6817A74884C_ServiceInterface" implementationRef="" name="">
    <bpmn2:operation id="_E486B83F-1225-436D-94D5-C6817A74884C_ServiceOperation" implementationRef="" name=""/>
  </bpmn2:interface>
  <bpmn2:process drools:adHoc="false" drools:packageName="org.project" drools:version="1.0" id="AddUserBasicService" isExecutable="true" name="BasicModel" processType="Public">
    <bpmn2:sequenceFlow id="_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B" sourceRef="_E486B83F-1225-436D-94D5-C6817A74884C" targetRef="_96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
      <bpmn2:extensionElements>
        <drools:metaData name="isAutoConnection.source">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
        <drools:metaData name="isAutoConnection.target">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
      </bpmn2:extensionElements>
    </bpmn2:sequenceFlow>
    <bpmn2:sequenceFlow id="_CF5E4B33-67E4-4424-9524-37114A55D4CA" sourceRef="_13126CFF-A8A6-4168-A236-C45A07DFF6DE" targetRef="_E486B83F-1225-436D-94D5-C6817A74884C">
      <bpmn2:extensionElements>
        <drools:metaData name="isAutoConnection.source">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
        <drools:metaData name="isAutoConnection.target">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
      </bpmn2:extensionElements>
    </bpmn2:sequenceFlow>
    <bpmn2:endEvent id="_96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
      <bpmn2:incoming>_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B</bpmn2:incoming>
    </bpmn2:endEvent>
    <bpmn2:serviceTask drools:serviceimplementation="Java" drools:serviceinterface="" drools:serviceoperation="" id="_E486B83F-1225-436D-94D5-C6817A74884C" implementation="Java" name="Add user to database" operationRef="_E486B83F-1225-436D-94D5-C6817A74884C_ServiceOperation">
      <bpmn2:extensionElements>
        <drools:metaData name="elementname">
          <drools:metaValue><![CDATA[Add user to database]]></drools:metaValue>
        </drools:metaData>
      </bpmn2:extensionElements>
      <bpmn2:incoming>_CF5E4B33-67E4-4424-9524-37114A55D4CA</bpmn2:incoming>
      <bpmn2:outgoing>_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B</bpmn2:outgoing>
    </bpmn2:serviceTask>
    <bpmn2:startEvent id="_13126CFF-A8A6-4168-A236-C45A07DFF6DE">
      <bpmn2:outgoing>_CF5E4B33-67E4-4424-9524-37114A55D4CA</bpmn2:outgoing>
    </bpmn2:startEvent>
  </bpmn2:process>
  <bpmndi:BPMNDiagram>
    <bpmndi:BPMNPlane bpmnElement="AddUserBasicService">
      <bpmndi:BPMNShape bpmnElement="_13126CFF-A8A6-4168-A236-C45A07DFF6DE" id="shape__13126CFF-A8A6-4168-A236-C45A07DFF6DE">
        <dc:Bounds height="56" width="56" x="577" y="247"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="_E486B83F-1225-436D-94D5-C6817A74884C" id="shape__E486B83F-1225-436D-94D5-C6817A74884C">
        <dc:Bounds height="102" width="154" x="713" y="224"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="_96CAC0E9-FA73-4154-84A6-A963ACD1ED04" id="shape__96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
        <dc:Bounds height="56" width="56" x="947" y="247"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="_CF5E4B33-67E4-4424-9524-37114A55D4CA" id="edge_shape__13126CFF-A8A6-4168-A236-C45A07DFF6DE_to_shape__E486B83F-1225-436D-94D5-C6817A74884C">
        <di:waypoint x="633" y="275"/>
        <di:waypoint x="713" y="275"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge bpmnElement="_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B" id="edge_shape__E486B83F-1225-436D-94D5-C6817A74884C_to_shape__96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
        <di:waypoint x="867" y="275"/>
        <di:waypoint x="947" y="275"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
  <bpmn2:relationship type="BPSimData">
    <bpmn2:extensionElements>
      <bpsim:BPSimData>
        <bpsim:Scenario id="default" name="Simulationscenario">
          <bpsim:ScenarioParameters/>
          <bpsim:ElementParameters elementRef="_13126CFF-A8A6-4168-A236-C45A07DFF6DE">
            <bpsim:TimeParameters>
              <bpsim:ProcessingTime>
                <bpsim:NormalDistribution mean="0" standardDeviation="0"/>
              </bpsim:ProcessingTime>
            </bpsim:TimeParameters>
          </bpsim:ElementParameters>
          <bpsim:ElementParameters elementRef="_E486B83F-1225-436D-94D5-C6817A74884C">
            <bpsim:TimeParameters>
              <bpsim:ProcessingTime>
                <bpsim:NormalDistribution mean="0" standardDeviation="0"/>
              </bpsim:ProcessingTime>
            </bpsim:TimeParameters>
            <bpsim:ResourceParameters>
              <bpsim:Availability>
                <bpsim:FloatingParameter value="0"/>
              </bpsim:Availability>
              <bpsim:Quantity>
                <bpsim:FloatingParameter value="0"/>
              </bpsim:Quantity>
            </bpsim:ResourceParameters>
            <bpsim:CostParameters>
              <bpsim:UnitCost>
                <bpsim:FloatingParameter value="0"/>
              </bpsim:UnitCost>
            </bpsim:CostParameters>
          </bpsim:ElementParameters>
        </bpsim:Scenario>
      </bpsim:BPSimData>
    </bpmn2:extensionElements>
    <bpmn2:source>_hImTMHArEDiY0b-88EQFvg</bpmn2:source>
    <bpmn2:target>_hImTMHArEDiY0b-88EQFvg]</bpmn2:target>
  </...> but was:<...w.jboss.org/drools" [xmlns:xsi="xsi" exporter="jBPM Process Modeler" exporterVersion="2.0" id="_otWKkDzdEDmaKPtB3c2pCA" targetNamespace="http://www.omg.org/bpmn20" xsi:schemaLocation="http://www.omg.org/spec/BPMN/20100524/MODEL BPMN20.xsd http://www.jboss.org/drools drools.xsd http://www.bpsim.org/schemas/1.0 bpsim.xsd http://www.omg.org/spec/DD/20100524/DC DC.xsd http://www.omg.org/spec/DD/20100524/DI DI.xsd">
  <bpmn2:interface id="_E486B83F-1225-436D-94D5-C6817A74884C_ServiceInterface" implementationRef="" name="">
    <bpmn2:operation id="_E486B83F-1225-436D-94D5-C6817A74884C_ServiceOperation" implementationRef="" name=""/>
  </bpmn2:interface>
  <bpmn2:process drools:adHoc="false" drools:packageName="org.project" drools:version="1.0" id="AddUserBasicService" isExecutable="true" name="BasicModel" processType="Public">
    <bpmn2:sequenceFlow id="_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B" sourceRef="_E486B83F-1225-436D-94D5-C6817A74884C" targetRef="_96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
      <bpmn2:extensionElements>
        <drools:metaData name="isAutoConnection.source">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
        <drools:metaData name="isAutoConnection.target">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
      </bpmn2:extensionElements>
    </bpmn2:sequenceFlow>
    <bpmn2:sequenceFlow id="_CF5E4B33-67E4-4424-9524-37114A55D4CA" sourceRef="_13126CFF-A8A6-4168-A236-C45A07DFF6DE" targetRef="_E486B83F-1225-436D-94D5-C6817A74884C">
      <bpmn2:extensionElements>
        <drools:metaData name="isAutoConnection.source">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
        <drools:metaData name="isAutoConnection.target">
          <drools:metaValue><![CDATA[true]]></drools:metaValue>
        </drools:metaData>
      </bpmn2:extensionElements>
    </bpmn2:sequenceFlow>
    <bpmn2:endEvent id="_96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
      <bpmn2:incoming>_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B</bpmn2:incoming>
    </bpmn2:endEvent>
    <bpmn2:serviceTask drools:serviceimplementation="Java" drools:serviceinterface="" drools:serviceoperation="" id="_E486B83F-1225-436D-94D5-C6817A74884C" implementation="Java" name="Add user to database" operationRef="_E486B83F-1225-436D-94D5-C6817A74884C_ServiceOperation">
      <bpmn2:extensionElements>
        <drools:metaData name="elementname">
          <drools:metaValue><![CDATA[Add user to database]]></drools:metaValue>
        </drools:metaData>
      </bpmn2:extensionElements>
      <bpmn2:incoming>_CF5E4B33-67E4-4424-9524-37114A55D4CA</bpmn2:incoming>
      <bpmn2:outgoing>_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B</bpmn2:outgoing>
    </bpmn2:serviceTask>
    <bpmn2:startEvent id="_13126CFF-A8A6-4168-A236-C45A07DFF6DE">
      <bpmn2:outgoing>_CF5E4B33-67E4-4424-9524-37114A55D4CA</bpmn2:outgoing>
    </bpmn2:startEvent>
  </bpmn2:process>
  <bpmndi:BPMNDiagram>
    <bpmndi:BPMNPlane bpmnElement="AddUserBasicService">
      <bpmndi:BPMNShape bpmnElement="_13126CFF-A8A6-4168-A236-C45A07DFF6DE" id="shape__13126CFF-A8A6-4168-A236-C45A07DFF6DE">
        <dc:Bounds height="56" width="56" x="577" y="247"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="_E486B83F-1225-436D-94D5-C6817A74884C" id="shape__E486B83F-1225-436D-94D5-C6817A74884C">
        <dc:Bounds height="102" width="154" x="713" y="224"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="_96CAC0E9-FA73-4154-84A6-A963ACD1ED04" id="shape__96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
        <dc:Bounds height="56" width="56" x="947" y="247"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="_CF5E4B33-67E4-4424-9524-37114A55D4CA" id="edge_shape__13126CFF-A8A6-4168-A236-C45A07DFF6DE_to_shape__E486B83F-1225-436D-94D5-C6817A74884C">
        <di:waypoint x="633" y="275"/>
        <di:waypoint x="713" y="275"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge bpmnElement="_945FC403-B19A-4AB1-9A9A-B13E6CD19D4B" id="edge_shape__E486B83F-1225-436D-94D5-C6817A74884C_to_shape__96CAC0E9-FA73-4154-84A6-A963ACD1ED04">
        <di:waypoint x="867" y="275"/>
        <di:waypoint x="947" y="275"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
  <bpmn2:relationship type="BPSimData">
    <bpmn2:extensionElements>
      <bpsim:BPSimData>
        <bpsim:Scenario id="default" name="Simulationscenario">
          <bpsim:ScenarioParameters/>
          <bpsim:ElementParameters elementRef="_13126CFF-A8A6-4168-A236-C45A07DFF6DE">
            <bpsim:TimeParameters>
              <bpsim:ProcessingTime>
                <bpsim:NormalDistribution mean="0" standardDeviation="0"/>
              </bpsim:ProcessingTime>
            </bpsim:TimeParameters>
          </bpsim:ElementParameters>
          <bpsim:ElementParameters elementRef="_E486B83F-1225-436D-94D5-C6817A74884C">
            <bpsim:TimeParameters>
              <bpsim:ProcessingTime>
                <bpsim:NormalDistribution mean="0" standardDeviation="0"/>
              </bpsim:ProcessingTime>
            </bpsim:TimeParameters>
            <bpsim:ResourceParameters>
              <bpsim:Availability>
                <bpsim:FloatingParameter value="0"/>
              </bpsim:Availability>
              <bpsim:Quantity>
                <bpsim:FloatingParameter value="0"/>
              </bpsim:Quantity>
            </bpsim:ResourceParameters>
            <bpsim:CostParameters>
              <bpsim:UnitCost>
                <bpsim:FloatingParameter value="0"/>
              </bpsim:UnitCost>
            </bpsim:CostParameters>
          </bpsim:ElementParameters>
        </bpsim:Scenario>
      </bpsim:BPSimData>
    </bpmn2:extensionElements>
    <bpmn2:source>_otWKkDzdEDmaKPtB3c2pCA</bpmn2:source>
    <bpmn2:target>_otWKkDzdEDmaKPtB3c2pCA]</bpmn2:target>
  </...></pre>
</details>

<details>
  <summary>Log 2 errors during compilation</summary>
  <pre>[ERROR] Process Info: id:eob, pkg:com.example, name:eob, version:1.0 
Node Info: id: name: 
Parser message: (null: 254, 31): cvc-complex-type.2.4.b: The content of element 'bpmn2:ioSpecification' is not complete. One of '{"http://www.omg.org/spec/BPMN/20100524/MODEL":inputSet, "http://www.omg.org/spec/BPMN/20100524/MODEL":outputSet}' is expected.
[ERROR] Process Info: id:eob, pkg:com.example, name:eob, version:1.0 
Node Info: id: name: 
Parser message: (null: 292, 31): cvc-complex-type.2.4.b: The content of element 'bpmn2:ioSpecification' is not complete. One of '{"http://www.omg.org/spec/BPMN/20100524/MODEL":inputSet, "http://www.omg.org/spec/BPMN/20100524/MODEL":outputSet}' is expected.
[ERROR] Process Info: id:eob, pkg:com.example, name:eob, version:1.0 
Node Info: id: name: 
Parser message: (null: 407, 31): cvc-complex-type.2.4.b: The content of element 'bpmn2:ioSpecification' is not complete. One of '{"http://www.omg.org/spec/BPMN/20100524/MODEL":inputSet, "http://www.omg.org/spec/BPMN/20100524/MODEL":outputSet}' is expected.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/kgaevski/.m2/repository/com/google/protobuf/protobuf-java/3.6.1/protobuf-java-3.6.1.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
</pre>
</details>

**Comment by romartin** [comment]
Thanks for the early review @hasys !

I'll check the selenium tests now... let's see :S

About the errors in logs, are you talking about the BC or kogito compilation? Also, are those errors really from this PR? If so, it may indicate that at some point (where processing the `bpmn2:ioSpecification` nodes, wherever they are) we're dong somethign wrong on the marshallers, and now that we're adding the `schemaLocation`, it just displays those, as it's validating against the XML schemas. Makes sense?

**Comment by romartin** [comment]
Added a new commit with "potentially" fixes the integration tests, although no way to make those tests run locally :(
I'd been looking at profiles and properties on the pom file, but still nothing. 
Any ideas @hasys  @domhanak ? 
Anyway let's see if the commit fixes those tests...
Thanks!

**Comment by romartin** [comment]
jenkins do fdb

**Comment by romartin** [comment]
jenkins retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/resources/org/kie/workbench/common/stunner/kogito/client/selenium/new-diagram.bpmn2 [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-  <bpmn2:process xmlns:drools="http://www.jboss.org/drools" id="new-file" drools:version="1.0" drools:packageName="com.example" drools:adHoc="false" name="new-file" isExecutable="true" processType="Public"/>
-  <bpmndi:BPMNDiagram xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI">
-    <bpmndi:BPMNPlane bpmnElement="default"/>
+<bpmn2:definitions id="_fd7ksT2NEDmijalwZkmLtw" xsi:schemaLocation="http://www.omg.org/spec/BPMN/20100524/MODEL BPMN20.xsd http://www.jboss.org/drools drools.xsd http://www.bpsim.org/schemas/1.0 bpsim.xsd http://www.omg.org/spec/DD/20100524/DC DC.xsd http://www.omg.org/spec/DD/20100524/DI DI.xsd " exporter="jBPM Process Modeler" exporterVersion="2.0" targetNamespace="http://www.omg.org/bpmn20" xmlns:bpmn2="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:bpsim="http://www.bpsim.org/schemas/1.0" xmlns:drools="http://www.jboss.org/drools" xmlns:xsi="xsi">
+  <bpmn2:process id="new-file" drools:packageName="com.example" drools:version="1.0" drools:adHoc="false" name="new-file" isExecutable="true" processType="Public"/>
+  <bpmndi:BPMNDiagram>
+    <bpmndi:BPMNPlane bpmnElement="new-file"/>
```
Comment (domhanak) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/resources/org/kie/workbench/common/stunner/kogito/client/selenium/new-diagram.bpmn2:5]: The issue is this element, you changed it to `new-file` but when you use editor to create new diagram it it set to `default`

**Comment by domhanak** [comment]
@romartin those tests are part of `kie-wb-common-stunner-bpmn-kogito-runtime` and there is profile named `kogito-it-tests`
so running `mvn install -Pkogito-it-tests` in the folder after you built whole stunner works. There is also a README in the module, that you can check.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/resources/org/kie/workbench/common/stunner/kogito/client/selenium/new-diagram.bpmn2 (hunk 1 above)
Comment (domhanak) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/resources/org/kie/workbench/common/stunner/kogito/client/selenium/new-diagram.bpmn2:5]: Test does this:
1. Open editor with empty content (create new file)
2. Get the content
3. Compare to a control instance

So it makes sense to update the control-instance to work with these changes.

**Comment by hasys** [comment]
> Thanks for the early review @hasys !
> 
> I'll check the selenium tests now... let's see :S
> 
> About the errors in logs, are you talking about the BC or kogito compilation? Also, are those errors really from this PR? If so, it may indicate that at some point (where processing the `bpmn2:ioSpecification` nodes, wherever they are) we're dong somethign wrong on the marshallers, and now that we're adding the `schemaLocation`, it just displays those, as it's validating against the XML schemas. If that case, this commit is just now raising the XML problems as errors, but the changes being applied are not the cause for these errors, so maybe would be better to create a new ticket and address it as a separate (we'll have to analyze which nodes are not properly being marshalled, as the schema declares, and so figure out how to fix).
> 
> Makes sense?

Hi @romartin, no those issues are visible before your fixes as well.

Looks like this PR wasn't ported from BC to Kogito: https://issues.redhat.com/browse/RHPAM-2388 so it is definitely should be fixed by separate PR.

Example:
* Output added in BC: https://github.com/kiegroup/kie-wb-common/pull/2964/files
* Output missing in Kogito: https://github.com/kiegroup/kie-wb-common/blob/80a34e97ae88541fcd884d76690f07fe11890479/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomInput.java#L96

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/resources/org/kie/workbench/common/stunner/kogito/client/selenium/new-diagram.bpmn2 (hunk 1 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/resources/org/kie/workbench/common/stunner/kogito/client/selenium/new-diagram.bpmn2:5]: right @domhanak , sorry my fault, just did it too fast... thanks :+1:

**Comment by romartin** [comment]
@hasys @domhanak thanks for your reviews and feedback guys! I'm working on the updates, will let you know

@hasys could you please report the issue you mention [above](https://github.com/kiegroup/kie-wb-common/pull/3544#issuecomment-764546800)?

**Comment by romartin** [comment]
Hey @hasys @domhanak 
Thanks for your feedback. I was finally able to run, also fix properly, the integration tests as well. So hopefully everything will be ok now. Ready for review again.
Thanks!

**Comment by romartin** [comment]
jenkins do fdb

**Review by hasys — COMMENTED** [review]
Great job, @romartin!

Failing test is random unrelated test, I think PR is ready to merge.

<details>
  <summary>Failed test</summary>
  <pre>
org.kie.bc.DeploymentIT.org.kie.bc.DeploymentIT

Error Message

Cannot deploy business-monitoring-webapp.war: {"WFLYCTL0062: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-1" => {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"business-monitoring-webapp.war\".WeldStartService" => "Failed to start service
    Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@6dee1b15
    Caused by: org.jboss.weld.exceptions.WeldException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@6dee1b15
    Caused by: java.lang.reflect.InvocationTargetException
    Caused by: org.jboss.weld.exceptions.WeldException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@4384e6a5
    Caused by: java.lang.reflect.InvocationTargetException
    Caused by: org.uberfire.java.nio.file.FileSystemNotFoundException: No filesystem for uri (default://system/system) found."}}}}

Stacktrace

org.jboss.arquillian.container.spi.client.container.DeploymentException: 
Cannot deploy business-monitoring-webapp.war: {"WFLYCTL0062: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-1" => {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"business-monitoring-webapp.war\".WeldStartService" => "Failed to start service
    Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@6dee1b15
    Caused by: org.jboss.weld.exceptions.WeldException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@6dee1b15
    Caused by: java.lang.reflect.InvocationTargetException
    Caused by: org.jboss.weld.exceptions.WeldException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@4384e6a5
    Caused by: java.lang.reflect.InvocationTargetException
    Caused by: org.uberfire.java.nio.file.FileSystemNotFoundException: No filesystem for uri (default://system/system) found."}}}}
</pre>
</details>

**Comment by hasys** [comment]
Issue we talked above reported as [KOGITO-4271](https://issues.redhat.com/browse/KOGITO-4271)

**Comment by romartin** [comment]
thanks  @hasys , if you're happy please approve.
@domhanak waiting for you feedback, thanks!

**Review by hasys — APPROVED** [review]

**Comment by romartin** [comment]
please @domhanak missing review for his blocker. Thanks

**Review by domhanak — APPROVED** [review]
LGTM.

**Comment by domhanak** [comment]
Yep, I saw the DeploymentIT failing on other PR's too.

**Pull request merged**